DATA_FORMAT.md

```diff
@@ -0,0 +1,372 @@
+  - 📘 feature_2.json
+  - 📁 frames/
+    - 📷 frame_0.png
+    - 📷 frame_1.png
+    - 📷 frame_2.png
+    ...
+    - 📷 frame_245.png
+```
+
+The `manifest.json` file would look something like this:
+
+```
+--manifest.json--
+{
+    "frames": [
+        "frames/frame_0.png",
+        "frames/frame_1.png",
+        "frames/frame_2.png",
+        ...
+        "frames/frame_245.png",
+    ],
+    "features": {
+        "My Cool Feature": "feature_0.json",
+        "Another Cool Feature": "feature_1.json",
+        "The Coolest Feature": "feature_2.json",
+    },
+    "tracks": "tracks.json",
+    "times": "times.json",
+    "outliers": "outliers.json",
+    "centroids": "centroids.json",
+    "bounds": "bounds.json"
+}
+```
+
+---
+
+</details>
+
+### 1. Tracks
+
+Every segmented object in each time step has an **object ID**, an integer identifier that is unique across all time steps. To recognize the same object across multiple frames, these object IDs must be grouped together into a **track** with a single **track number/track ID**.
+
+A **track JSON file** consists of a JSON object with a `data` array, where for each object ID `i`, `data[i]` is the track number that object is assigned to.
+
+```
+--track.json--
+{
+    "data": [
+        <track number for id 0>,
+        <track number for id 1>,
+        <track number for id 2>,
+        ...
+    ]
+}
+```
+
+<details>
+<summary><b>[Show me an example!]</b></summary>
+
+---
+
+For example, if there were the following two tracks in some dataset, the track file might look something like this.
+
+| Track # | Object IDs |
+| ------- | ---------- |
+| 1       | 0, 1, 4    |
+| 2       | 2, 3, 5    |
+
+Note that the object IDs are not guaranteed to be sequential!
+
+```
+--track.json--
+{
+    "data": [
+        1, // 0
+        1, // 1
+        2, // 2
+        2, // 3
+        1, // 4
+        2  // 5
+    ]
+}
+```
+
+---
+
+</details>
+
+### 2. Times
+
+The times JSON is similar to the tracks JSON. It also contains a `data` array that maps from object IDs to the frame number that they appear on.
+
+```
+--times.json--
+{
+    "data": [
+        <frame number for id 0>,
+        <frame number for id 1>,
+        <frame number for id 2>,
+        ...
+    ]
+}
+```
+
+### 3. Frames
+
+_Example frame:_
+![](./documentation/frame_0.png)
+_Each unique color in this frame is a different object ID._
+
+**Frames** are image textures that store the object IDs for each time step in the time series. Each pixel in the image can encode a single object ID in its RGB value (`object ID = R + G*256 + B*256*256 - 1`), and background pixels are `#000000` (black).
+
+Additional notes:
+
+- There is currently no way to indicate overlaps in the data-- your script will need to flatten 3D data and segmentations.
+- Encoded object ID's in the frame data start at `1` instead of `0`, because `#000000` (black) is reserved for the background.
+- The highest object ID that can currently be represented is `16,843,007`.
```

Comment:
This is an interesting limitation to note, and I might also stress that we can remove that limit if the _total number of segmented objects for an entire time series_ ever exceeds it.